perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -112,6 +112,8 @@
 	final static Logger log = LoggerFactory.getLogger(MembersManagerBlImpl.class);
 
 	private static final String EXPIRATION = AttributesManager.NS_MEMBER_ATTR_DEF + ":membershipExpiration";
+	private static final String EXPIRATION_RULES_ATTR = AttributesManager.NS_VO_ATTR_DEF + ":membershipExpirationRules";
+	private static final String SPONSORED_MEMBERS_EXPIRATION_KEY = "doNotExpireSponsoredMembers";
```

Comment:
Are you sure this is the right place for these variables? Can't you use a module for attribute instead? By this, you are creating a duplicate of the key in the code.

Reply:
I have used static variables from module urn_perun_vo_attribute_def_def_membershipExpirationRules.